docs/source/java/quickstartguide.rst

```diff
@@ -0,0 +1,325 @@
+
+Example: create a field named "document" of string type:
+
+.. code-block:: Java
+
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import java.util.HashMap;
+    import java.util.Map;
+
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put("A", "Id card");
+    metadata.put("B", "Passport");
+    metadata.put("C", "Visa");
+    Field document = new Field("document",
+            new FieldType(true, new ArrowType.Utf8(), /*dictionary*/ null, metadata),
+            /*children*/ null);
+    System.out.println("Field created: " + document + ", Metadata: " + document.getMetadata());
+
+.. code-block:: shell
+
+    Field created: document: Utf8, Metadata: {A=Id card, B=Passport, C=Visa}
+
+Create a Schema
+***************
+
+**Schema** holds a sequence of fields together with some optional metadata.
+
+Example: Create a schema describing datasets with two columns:
+a int32 column "A" and a utf8-encoded string column "B"
+
+.. code-block:: Java
+
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+    import java.util.HashMap;
+    import java.util.Map;
+    import static java.util.Arrays.asList;
+
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put("K1", "V1");
+    metadata.put("K2", "V2");
+    Field a = new Field("A", FieldType.nullable(new ArrowType.Int(32, true)), /*children*/ null);
+    Field b = new Field("B", FieldType.nullable(new ArrowType.Utf8()), /*children*/ null);
+    Schema schema = new Schema(asList(a, b), metadata);
+    System.out.println("Schema created: " + schema);
+
+.. code-block:: shell
+
+    Schema created: Schema<A: Int(32, true), B: Utf8>(metadata: {K1=V1, K2=V2})
+
+Create a VectorSchemaRoot
+*************************
+
+**VectorSchemaRoot** combines ValueVectors with a Schema to represent tabular data.
+
+Example: Create a dataset with metadata that contains integer age and
+string names of data.
+
+.. code-block:: Java
+
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.IntVector;
+    import org.apache.arrow.vector.VarCharVector;
+    import org.apache.arrow.vector.VectorSchemaRoot;
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+    import java.nio.charset.StandardCharsets;
+    import java.util.HashMap;
+    import java.util.Map;
+    import static java.util.Arrays.asList;
+
+    Field a = new Field("age",
+            FieldType.nullable(new ArrowType.Int(32, true)),
+            /*children*/null
```

Comment:
Does Field require you to pass children?

Reply:
Yes, it is required